classes/class.ilObjShortLink.php

```diff
@@ -0,0 +1,303 @@
+        $shortLinks = array();
+        $currentUser = $this->usr->getLogin();
+
+        $set = $this->db->query('SELECT * FROM ' . ilShortLinkPlugin::TABLE_NAME);
+        $isAdministrator = $this->checkAdministrationPrivilegesFromDB();
+        while ($rec = $this->db->fetchAssoc($set)) {
+            if ($as_obj) {
+                if($currentUser == $rec['contact_user_login'] || $isAdministrator){
+                    $shortLinks[] = array('id'=> (int)$rec['id'], 'long_url'=>$rec['full_url'], 'short_link'=>$rec['short_link'],
+                        'customer'=>$rec['customer'], 'contact'=>$rec['contact_user_login']);
+                }
+            } else {
+                $shortLinks[] = $rec;
+            }
+        }
+        return $shortLinks;
+    }
+
+    public function doUpdate() {
+        $this->db->manipulate('UPDATE ' . ilShortLinkPlugin::TABLE_NAME . ' SET' .
+            ' short_link = ' . $this->db->quote($this->getShortLink(), 'text') . ',' .
+            ' full_url = ' . $this->db->quote($this->getLongURL(), 'text') . ',' .
+            ' customer = ' . $this->db->quote($this->getCustomer(), 'text') .
+            ' WHERE id = ' . $this->db->quote($this->getId(), 'text') . ';'
+        );
+    }
+
+    /**
+     * Get user login name
+     *
+     * @param $idNum
+     * @return mixed
+     */
+    public function getOwner($idNum) {
+        $this->setId($idNum);
+        $set = $this->db->query('SELECT contact_user_login FROM ' . ilShortLinkPlugin::TABLE_NAME . ' WHERE id=' . $this->getId());
+        $rec = $this->db->fetchAssoc($set);
+        return $rec['contact_user_login'];
+    }
+
+    /**
+     * Gets longURL from shortURL
+     *
+     * @param $shortLink
+     * @return mixed
+     */
+    public function fetchLongURL($shortLink) {
+        $set = $this->db->query('SELECT full_url FROM ' .ilShortLinkPlugin::TABLE_NAME . ' WHERE short_link=' . "'" . $shortLink . "'");
+        $rec = $this->db->fetchAssoc($set);
+        return $rec['full_url'];
+    }
+
+    public function checkIfShortLinkAlreadyMentioned($shortLink) {
+        $set = $this->db->query('SELECT full_url FROM ' .ilShortLinkPlugin::TABLE_NAME . ' WHERE short_link=' . "'" . $shortLink . "'");
+        $rec = $this->db->fetchAssoc($set);
+        if($rec == NULL)  {
+            return false;
+        }
+        return true;
+    }
+
+
+    /**
+     *
+     * Deletes the entry with id $id from DB
+     *
+     * @param $id
+     */
+    public function doDelete($id) {
+        $this->db->manipulate('DELETE FROM ' . ilShortLinkPlugin::TABLE_NAME .
+            ' WHERE id = ' . $id, 'integer');
+    }
+
+    /**
+     * Returns true if Administrator privilege is granted
+     *
+     * @return bool
+     */
+    public function checkAdministrationPrivilegesFromDB() {
+        $administrationRole = $this->getRoleIdOfAdministrator();
+        if($administrationRole == -1) {
```

Comment:
The function promises a check and should return true or false, not perform redirects.